you-should-use.plugin.zsh

```diff
@@ -1,14 +1,52 @@
 #!/bin/zsh
 
 function ysu_message() {
+  local BOLD='\033[1m'
+  local NONE='\033[00m'
   echo "${BOLD}Found existing alias for \"$1\". You should use: \"$2\"${NONE}"
 }
 
 
-function _check_aliases() {
+function ysu_global_message() {
   local BOLD='\033[1m'
-  local RED='\e[31m'
   local NONE='\033[00m'
+  echo "${BOLD}Found existing global alias for \"$1\". You should use: \"$2\"${NONE}"
+}
+
+
+# Prevent command from running if hardcore mode enabled
+function _check_ysu_hardcore() {
+  if [[ "$YSU_HARDCORE" = 1 ]]; then
+      local RED='\e[31m'
+      echo "${BOLD}${RED}You Should Use hardcore mode enabled. Use your aliases!${NONE}"
+      kill -s INT $$
+  fi
+}
+
+
+function _check_global_aliases() {
+  IFS="\n"
+  local global_aliases="$(alias -g)"
+  local found=""
+  for entry in $global_aliases; do
```

Comment:
I've just realise my use of `tr` is also probably less than ideal, you'll also need to consider these edge cases:

```zsh
alias -g _quot="'"`
alias -g  # _quot=\'
```

```zsh
alias -g _quot="this is a single quote: '"
alias -g  # _quot='this is a single quote: '\'
```